api/envoy/config/response_map/v3/response_map.proto

```diff
@@ -0,0 +1,89 @@
+syntax = "proto3";
+
+package envoy.config.response_map.v3;
+
+import "envoy/config/accesslog/v3/accesslog.proto";
+import "envoy/config/core/v3/base.proto";
+import "envoy/config/core/v3/substitution_format_string.proto";
+
+import "google/protobuf/struct.proto";
+import "google/protobuf/wrappers.proto";
+
+import "udpa/annotations/status.proto";
+import "udpa/annotations/versioning.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.config.response_map.v3";
+option java_outer_classname = "ResponseMapProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: ResponseMap]
+// Response map filter :ref:`configuration overview <config_response_map>`.
+
+// The configuration to filter and change HTTP responses.
+// [#next-free-field: 6]
+message ResponseMapper {
+  // Filter to determine if this mapper should apply.
+  accesslog.v3.AccessLogFilter filter = 1 [(validate.rules).message = {required: true}];
```

Comment:
@lizan can I ask you to do an API pass before I dig into the code?
I think a lot of the components of access logs apply but not all do - not sure how much API refactor we want to do here and how much we can just do via standard metadata matcher.

Reply:
@snowp can you weigh in here re: the new matcher API fits here? I think the access log filter works well for this purpose but just want to double check.

Reply:
I would think that this could be accomplished with the new matching API, though we would most likely spend some time to reach parity in terms of supporting all the same conditionals that the access log filter supports. The matching API would offer some benefit in terms of better performance when lots of matchers are used, due to being able to structure it more as a tree than as a list of complex matchers. 

I think this falls into the category that we'd use the matching API if it was available today but since that work is ongoing it becomes a question of time lines. Maybe @mattklein123 has thoughts on whether we're okay with relying on the access log filters here.

I could imagine the access log filter eventually being subsumed by the new matching work, but that would be a longer term thing

Reply:
Yeah, I'd initially hoped the new matcher code could be used, since using access filters just strikes me as suboptimal and I'd like the perf wins.  Looks like it'd also do away with much of the shared code, so probably the way to go (if we opt to have a standalone filter for per-route) is stick with this for now, and move both the HCM and filter matchers over to the new API when it's more feature complete.